File: Web/Tushar-sahani_TusharSahani_2125cs1072_2/FunctionAndArrays/RotateAnArray.java
package coderspear.FunctionAndArrays;
import java.io.*;
public class RotateAnArray {
    public static void display(int[] a){
        StringBuilder sb = new StringBuilder();
    
        for(int val: a){
          sb.append(val + " ");
        }
        System.out.println(sb);
      }
    
      public static void rotate(int[] a, int k){
          int n = a.length;
          k = k%n;
          if(k<0){
            k = n+k;  
          }
          int[] arr = new int[n];
          for(int i=0;i<n;i++){
              arr[(i+k)%n] = a[i];
          }
          
          for(int i = 0;i < n;i++){
              a[i] = arr[i];
          }
      }
    
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    
        int n = Integer.parseInt(br.readLine());
        int[] a = new int[n];
        for(int i = 0; i < n; i++){
           a[i] = Integer.parseInt(br.readLine());
        }
        int k = Integer.parseInt(br.readLine());
    
        rotate(a, k);
        display(a);
     }
}

File: IOT/gurpreetk12_gurpreetkaur_2125cs1094_2/FunctionandArrays/RotateAnArray.cpp
#include<iostream>
using namespace std;

void rotate(int* arr, int n, int k){
    int r=k;
   if(r>0)
   {
       if(r<n)
       {for(int i=n-k;i<n;i++)
       {
           cout<<arr[i]<<" ";
       }
       for(int i=0;i<n-k;i++)
       {
           cout<<arr[i]<<" ";
       }}
       else
       {
           k=r%n;
           for(int i=n-k;i<n;i++)
       {
           cout<<arr[i]<<" ";
       }
       for(int i=0;i<n-k;i++)
       {
           cout<<arr[i]<<" ";
       }
       }
   }
   else
   {    k=k*-1;
        if(k<n)
       {for(int i=k;i<n;i++)
       {
           cout<<arr[i]<<" ";
       }
       for(int i=0;i<k;i++)
       {
           cout<<arr[i]<<" ";
       }}
       else
       {
           k=k%n;
           for(int i=k;i<n;i++)
       {
           cout<<arr[i]<<" ";
       }
       for(int i=0;i<k;i++)
       {
           cout<<arr[i]<<" ";
       }
           
       }
   }
  
    
}



int main(){
    int n, r;
    cin>>n;
    int* arr = new int[n];
    for(int i = 0 ; i < n; i++){
        cin>>arr[i];
    }
    cin>>r;
    
    rotate(arr,n,r);
    //display(arr,n);
}